functions/Start-DbaXESession.ps1

```diff
@@ -0,0 +1,133 @@
+function Start-DbaXESession {
+<#
+	.SYNOPSIS
+	Starts Extended Events sessions.
+
+	.DESCRIPTION
+	This script starts Extended Events sessions on a SQL Server instance.
+
+	.PARAMETER SqlInstance
+	The SQL Instances that you're connecting to.
+
+	.PARAMETER SqlCredential
+	Credential object used to connect to the SQL Server as a different user
+
+	.PARAMETER Session
+	Only start specific Extended Events sessions.
+
+	.PARAMETER AllSessions
+	Start all Extended Events sessions on an instance, ignoring the packaged sessions: AlwaysOn_health, system_health, telemetry_xevents.
+
+	.PARAMETER Silent
+	If this switch is enabled, the internal messaging functions will be silenced.
+
+	.NOTES
+	Tags: Memory
+	Author: Doug Meyers
+	Website: https://dbatools.io
+	Copyright: (C) Chrissy LeMaire, clemaire@gmail.com
+	License: GNU GPL v3 https://opensource.org/licenses/GPL-3.0
+
+	.LINK
+	https://dbatools.io/Start-DbaXESession
+
+	.EXAMPLE
+	Start-DbaXESession -SqlInstance sqlserver2012 -AllSessions
+
+	Starts all Extended Event Session on the sqlserver2014 instance.
+
+	.EXAMPLE
+	Start-DbaXESession -SqlInstance sqlserver2012 -Session xesession1,xesession2
+
+	Starts the xesession1 and xesession2 Extended Event sessions.
+
+	.EXAMPLE
+	Get-DbaXESession -SqlInstance sqlserver2012 -Session xesession1 | Start-DbaXESession
+
+	Starts the sessions returned from the Get-DbaXESession function.
+
+#>
+	[CmdletBinding(DefaultParameterSetName='Session')]  
+	param (
+		[parameter(Position=1, Mandatory, ParameterSetName='Session')]
+		[parameter(Position=1, Mandatory, ParameterSetName='All')]
+		[Alias("ServerInstance", "SqlServer")]
+		[DbaInstanceParameter[]]$SqlInstance,
+
+		[parameter(ParameterSetName='Session')]
+		[parameter(ParameterSetName='All')]
+		[PSCredential]$SqlCredential,
+
+		[parameter(Mandatory, ParameterSetName='Session')]
+		[Alias("Sessions")]
+		[object[]]$Session,
+
+		[parameter(Mandatory, ParameterSetName='All')]
+		[switch]$AllSessions,
```

Comment:
This is a dangerous parameter to add when it comes to XE and I would disagree with adding it. Unless @potatoqualitee has some overwhelming evidence to think it needs to be there.

Starting up EVERY session on a given instance should never be done in the real world because the DBA does not know what those sessions are doing. An example, you have a one-off session configured that was capturing execution plans on an OLTP system. A session that was capturing AG related information (see examples from TigerTeam) that is recommended by Microsoft to never be left running for an extended period of time.

I don't think this should be offered...but that is just my opinion.

Reply:
I think there are cases when it'd be used. We could add some additional prompts to if they use it